src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -5199,6 +5199,36 @@ private BoundExpression BindCollectionExpression(CollectionExpressionSyntax synt
 
             MessageID.IDS_FeatureCollectionExpressions.CheckFeatureAvailability(diagnostics, syntax, syntax.OpenBracketToken.GetLocation());
 
+            for (int i = 1, n = syntax.Elements.Count - 1; i < n; i++)
+            {
+                var element = syntax.Elements[i];
+                if (element is WithElementSyntax withElement)
+                {
+                    Error(diagnostics, ErrorCode.ERR_WithElementMustBeFirst, withElement.WithKeyword.GetLocation());
+                    break;
+                }
+            }
+
+            foreach (var element in syntax.Elements)
+            {
+                if (element is KeyValuePairElementSyntax keyValuePairElement)
+                {
+                    MessageID.IDS_FeatureDictionaryExpressions.CheckFeatureAvailability(diagnostics, syntax, keyValuePairElement.ColonToken.GetLocation());
+
+                    // PROTOTYPE: Error for now.  Flesh this out when we do the binding for kvp elements.
+                    Error(diagnostics, ErrorCode.ERR_SyntaxError, keyValuePairElement.ColonToken, ",");
+                    return new BoundBadExpression(syntax, LookupResultKind.Empty, symbols: [], childBoundNodes: [], CreateErrorType());
+                }
+                else if (element is WithElementSyntax withElement)
+                {
+                    MessageID.IDS_FeatureCollectionExpressionConstructionArguments.CheckFeatureAvailability(diagnostics, syntax, withElement.WithKeyword.GetLocation());
+
+                    // PROTOTYPE: Error for now.  Flesh this out when we do the binding for `with(...)` arguments.
+                    Error(diagnostics, ErrorCode.ERR_SyntaxError, withElement.WithKeyword, ",");
+                    return new BoundBadExpression(syntax, LookupResultKind.Empty, symbols: [], childBoundNodes: [], CreateErrorType());
```

Comment:
>return new BoundBadExpression [](http://example.com/codeflow?start=20&length=29)

Is there a reason to fail binding for the collection, or even the element?

Reply:
this was just to unblock the PR.  i would expect us to want us to be very resilient in practice.  but i did no binding work.